packages/contracts-periphery/contracts/universal/DrippieRedd.sol

```diff
@@ -0,0 +1,226 @@
+        // string to be used for balance threshold checks. Specifically, we look for a 33 byte
+        // string starting with the 0x00 (STOP) opcode, followed by the uint256 threshold amount.
+        // Since a leading STOP opcode would normally halt execution (and be a useless checkscript)
+        // we can safely treat this as a special string. Saves ~30k gas.
+        bool executable;
+        if (checkscript[0] == hex"00" && checkscript.length == 33) {
+            assembly {
+                let threshold := mload(add(checkscript, 33))
+                executable := lt(balance(recipient), threshold)
+            }
+        } else {
+            // Checkscript is only part of the EVM bytecode that actually gets executed on-chain.
+            // We prepend a bytecode snippet that pushes the recipient address onto the stack and
+            // allows the checkscript to operate on it. We then also append a snippet that takes
+            // the final value on the stack and stores it in memory at 0..32 before reverting with
+            // that value.
+            bytes memory script = abi.encodePacked(
+                // Snippet for pushing the recipient address to the stack.
+                hex"73",
+                recipient,
+                // Actual user checkscript.
+                checkscript,
+                // Checkscript must leave a value on the stack, this cleanup segment will store
+                // that value into memory at position 0..32 and then revert with that value which
+                // allows us to access that value via returndatacopy below. This is a convenience
+                // since checkscripts would otherwise have to do this manually.
+                hex"60005260206000FD"
+            );
+
+            assembly {
+                // Create a contract using the checkscript as initcode. This will execute the EVM
+                // instructions included within the checkscript and, hopefully, leave a single 32
+                // byte returndata value. We don't actually care about the returned contract
+                // address since the script is intended to revert.
+                pop(create(0, add(script, 32), mload(script)))
+
+                // We expect the returned data to be exactly 32 bytes. Anything other than 32 bytes
+                // will be ignored, which means "executable" will remain false.
+                if eq(returndatasize(), 32) {
+                    let ret := mload(0x40)
+                    mstore(0x40, add(ret, 32))
+                    returndatacopy(ret, 0, 32)
+                    executable := eq(mload(ret), 1)
+                }
+            }
+        }
+
+        require(
+            executable == true,
+            "DrippieRedd: checkscript failed so drip is not yet ready to be triggered"
+        );
+
+        state.last = block.timestamp;
+        (bool success, ) = recipient.call{ value: state.config.amount }(state.config.data);
+
```

Comment:
This follows checks, effects, interactions so it should be safe from being re-entered